src/panels/developer-tools/service/developer-tools-service.js

```diff
@@ -284,7 +284,15 @@ class HaPanelDevService extends LocalizeMixin(PolymerElement) {
     return parsedJSON === ERROR_SENTINEL ? "" : parsedJSON.entity_id;
   }
 
-  _computeEntityDomainFilter(domain) {
+  _computeEntityDomainFilter(domain, attributes) {
+    const field = attributes.find((attribute) => attribute.key === "entity_id");
+    if (
+      field &&
+      "limit_entity_domain" in field &&
+      field.limit_entity_domain.length > 0
```

Comment:
you can change this to `if (field?.limit_entity_domain)` it will will be falsy if it is empty or don't exist.

Reply:
So this will work if field doesn't exist, limit_entity_domain doesn't exist as a field property, or limit_entity_domain is an empty list? Having trouble understanding null propagation or finding a good environment to test it in. I made the changes as suggested though

Reply:
Only thing it doesn't do is check for an empty list, but is that really necessary?

Reply:
Otherwise make it:
`if (field?.limit_entity_domain && field.limit_entity_domain.length)

Reply:
Oh my bad 😅 this is not a typescript file, `?` is a typescript function.

Reply:
```js
if (field && field.limit_entity_domain && field.limit_entity_domain.length)
```

Reply:
:smile: that explains while I didn’t understand how this works nor could I get it to work while testing. Will submit a revert/update later